packages/trpc/server/routers/viewer.tsx

```diff
@@ -374,9 +374,24 @@ const loggedInViewerRouter = router({
       destinationCalendarEmail = destinationCal?.email ?? user.destinationCalendar?.externalId;
     }
 
+    let destinationCalendarName = undefined;
+
+    for (const integration of connectedCalendars) {
+      if (integration?.calendars) {
+        for (const calendar of integration?.calendars) {
+          if (calendar.externalId === user.destinationCalendar?.externalId)
+            destinationCalendarName = calendar.name;
+        }
+      }
+    }
```

Comment:
I smell a code smell here. We're traversing everything just to attach a calendar name. I think we can do better here. Can we include this from the initial prisma query?

Reply:
We don't store the calendar name in the DB. I was thinking about saving the calendar name to the DB, but it could change anytime. By grabbing the calendar name in the connected calendars call, we can always ensure that the calendar name is up to date. 

Although looking at this now, we should only traverse through the connected calendars if they are the same integration as the destination calendar.